app/code/Magento/SalesRule/Model/Rule/DataProvider.php

```diff
@@ -60,17 +55,16 @@ public function __construct(
         $primaryFieldName,
         $requestFieldName,
         CollectionFactory $collectionFactory,
-        \Magento\Framework\Registry $registry,
-        \Magento\SalesRule\Model\Rule\Metadata\ValueProvider $metadataValueProvider,
+        Registry $registry,
```

Comment:
Why not promote this property?
```suggestion
        protected readonly Registry $coreRegistry,
```